### ISSUE TITLE
Find replacement for Docker Hub in CI pipeline

### ISSUE THREAD

**Issue by dnnr** [herbstluftwm/herbstluftwm#1334]
Docker has just announced that they'll shut down the autobuild feature for users on the free plan on June 18. Despite our switch to GitHub Actions, we're still using Docker Hub to build and persist an up-to-date Docker image (as far as I can tell, at least).

I'm assuming this setup with continue to work past the 18th for as long as the Dockerfiles don't need to be updated.

That said, we should find a replacement for this part of the CI pipeline. GitHub's offering might be a candidate, but there seems to be a 500MB storage cap. With some minimization efforts we might be able push our images below that. (focal is 372MB, trusty is 388MB)

**Comment by dnnr** [comment]
Update: Thorsten is applying for Docker's FOSS program so that we'll be exempt from this change.

**Comment by t-wissmann** [comment]
Let's see what happens:
> Thank you for completing the Docker Open Source Survey.  We will review your application, and get back to you as quickly as we can.  We have received a large of applications, so please bear with us, as we go through the review process

**Comment by t-wissmann** [comment]
I didn't receive any reaction so far. Just to make sure that I get it correctly what shutting down the *autobuild* feature actually means: does it mean we can still have our docker images there but only need to locally build them and push them manually whenver we change something?

**Comment by dnnr** [comment]
Right, pushing them manually would be a workaround for that. Oh, and I just realized: It shouldn't even break if we change a Dockerfile after the deadline, it should just increase the build time. The CI scripting ensures at run time that the pulled image fits the current Dockerfile, so if the hub image is out of date (because no autobuild), it will simply rebuild it ad-hoc.

**Comment by t-wissmann** [comment]
> The CI scripting ensures at run time that the pulled image fits the current Dockerfile, so if the hub image is out of date (because no autobuild), it will simply rebuild it ad-hoc.

Actually, the CI scripting does not do this anymore. It was the case for travis, but it's not true for github actions anymore.

**Comment by t-wissmann** [comment]
The assessment by docker will take some more time. I've just received an e-mail:


> Thank you for submitting your application to the Docker Open Source Programg
> and apologies for the delay in responding to you.  We are excited to have
> you as a part of our tremendously talented developer community, and weg
> appreciate that you took the time to fill out this application.  We are in
> the process of reviewing your application, and we will get back to you as
> soon as we have any updates.  Please bear with us as we go through this
> process, since this program has generated a lot of excitement, and we areg
> moving through the review process as quickly as we can.g
> Docker New Image Retention and Data Egress Policies

**Comment by blarz** [comment]
It's been a couple of months now, any updates?

**Comment by t-wissmann** [comment]
No there weren't updates. I think I missed a hidden approval step. in the above cited mail, a couple of paragraphs later, there was a sentence:

> \*Please reply to this email if your open source project complies with the                                                          
above criteria, and you would like to move forward with the review and                                                             
approval of your Open Source Program.\* 

So I've just replied to this e-mail. Let us see what happens.
